### PR TITLE
Template tile for wear OS

### LINKED ISSUE #1897: Template tile in Wear OS
**Is your feature request related to a problem? Please describe.**
No problem

**Describe the solution you'd like**
Have a tile that displays a template

**Describe alternatives you've considered, if any**
N.a.

**Additional context**

### PR DESCRIPTION
**Pull request by leroyboerefijn**
## Summary

This PR adds a template tile to Wear OS, to fix #1897. Changing the template is only possible on the phone app, since text input is difficult on Wear (no { and } on Gboard, no _ on draw input, small screen, etc).

## Screenshots

Tile:
Screenshot_20220103_230321_sysui

Wear settings:
Screenshot_20220106_143913_debug
Screenshot_20220106_144933_debug

Phone settings:
Screenshot_20220106-143843

## Link to pull request in Documentation repository

Documentation: home-assistant/companion.home-assistant#651

### CODE REVIEW

**Comment by dshokouhi** [comment]
OK got a chance to look at the phone and wear APK. I tried to see what would happen if I had a bunch of text on screen and it doesnt seem like the tile had a scroll bar.  Do tiles have a scroll bar or should we just truncate after X amount of characters? Not sure how we want to handle it but something to consider 


Regarding the phone side. Should we mimic what we do with the template widget and render it on the phone so the user can see the template in real time?

**Comment by leroyboerefijn** [comment]
> Do tiles have a scroll bar or should we just truncate after X amount of characters? Not sure how we want to handle it but something to consider

No, there's no scrollbars for tiles unfortunately. So maybe we need to truncate indeed. But the number of characters also depends on the screen size and amount of new lines right? So not sure how to deal with it...

> Regarding the phone side. Should we mimic what we do with the template widget and render it on the phone so the user can see the template in real time?

Great idea! I'll try to add something like that

**Comment by dshokouhi** [comment]
> No, there's no scrollbars for tiles unfortunately. So maybe we need to truncate indeed. But the number of characters also depends on the screen size and amount of new lines right? So not sure how to deal with it...

It might just be best for us to mention it in the docs that there are no scroll bars

**Comment by leroyboerefijn** [comment]
Preview added to the phone settings. Changing the docs PR now :)

File: common/src/main/res/values/strings.xml [hunk 1]
```diff
@@ -593,6 +594,13 @@
     <string name="successful">Successful</string>
     <string name="switches">Switches</string>
     <string name="tag_reader_title">Processing Tag</string>
+    <string name="template_tile">Template tile</string>
+    <string name="template_tile_change_message">Change template in phone settings</string>
+    <string name="template_tile_content">Template tile content</string>
+    <string name="template_tile_desc">Renders and displays a template</string>
+    <string name="template_tile_empty">Set template in the phone settings</string>
+    <string name="template_tile_error">Error in template</string>
+    <string name="template_tile_help">Provide a template below that will be displayed on the wear OS template tile.</string>
```
Comment (jpelgrom) [inline common/src/main/res/values/strings.xml:603]: Minor detail: wear OS should be Wear OS

File: app/src/full/java/io/homeassistant/companion/android/settings/wear/views/SettingsWearTemplateTile.kt [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+                            contentDescription = stringResource(id = R.string.help)
+                        )
+                    }
+                }
+            )
+        }
+    ) {
+        Column(Modifier.padding(top = 8.dp, start = 8.dp, end = 8.dp)) {
+            Row(verticalAlignment = Alignment.CenterVertically) {
+                Image(
```
Comment (jpelgrom) [inline app/src/full/java/io/homeassistant/companion/android/settings/wear/views/SettingsWearTemplateTile.kt:69]: This icon is invisible/white-on-white for me in light theme.

**Review by jpelgrom — COMMENTED** [review]
The tile is looking good! I've added a few comments to make the settings look more polished.

File: app/src/full/java/io/homeassistant/companion/android/settings/wear/views/SettingsWearTemplateTile.kt [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    template: String,
+    renderedTemplate: String,
+    refreshInterval: Int,
+    onContentChanged: (String) -> Unit,
+    onRefreshIntervalChanged: (Int) -> Unit
+) {
+    val context = LocalContext.current
+    Scaffold(
+        topBar = {
+            TopAppBar(
```
[the remark this answers is not in the document]
Comment (leroyboerefijn) [reply app/src/full/java/io/homeassistant/companion/android/settings/wear/views/SettingsWearTemplateTile.kt:51]: Good one. I'll add it to this view. Later I guess we need this for the other settings views as well. I propose to combine that with a bit of a redesign to match the main settings (non-compose)

File: app/src/full/java/io/homeassistant/companion/android/settings/wear/views/SettingsWearTemplateTile.kt [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+                title = { Text(stringResource(R.string.template_tile_content)) },
```
[the remark this answers is not in the document]
Comment (leroyboerefijn) [reply app/src/full/java/io/homeassistant/companion/android/settings/wear/views/SettingsWearTemplateTile.kt:52]: Agreed

File: wear/src/main/java/io/homeassistant/companion/android/home/views/SettingsView.kt [hunk 5]
```diff
@@ -200,6 +201,24 @@ fun SettingsView(
...
+                        icon = {
+                            Image(
+                                asset = CommunityMaterial.Icon3.cmd_text_box,
+                                colorFilter = ColorFilter.tint(wearColorPalette.onSurface)
+                            )
+                        },
+                        colors = ChipDefaults.secondaryChipColors(),
+                        label = {
+                            Text(
+                                text = stringResource(id = commonR.string.template_tile_content)
```
[the remark this answers is not in the document]
Comment (leroyboerefijn) [reply wear/src/main/java/io/homeassistant/companion/android/home/views/SettingsView.kt:216]: Agreed

File: app/src/full/java/io/homeassistant/companion/android/settings/wear/views/SettingsWearTemplateTile.kt [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+                        Icon(
+                            Icons.Filled.HelpOutline,
+                            contentDescription = stringResource(id = R.string.help)
+                        )
+                    }
+                }
+            )
+        }
+    ) {
+        Column(Modifier.padding(top = 8.dp, start = 8.dp, end = 8.dp)) {
```
[the remark this answers is not in the document]
Comment (leroyboerefijn) [reply app/src/full/java/io/homeassistant/companion/android/settings/wear/views/SettingsWearTemplateTile.kt:67]: You're right, good catch!

File: app/src/full/java/io/homeassistant/companion/android/settings/wear/views/SettingsWearTemplateTile.kt (hunk 2 above)
Comment (leroyboerefijn) [reply app/src/full/java/io/homeassistant/companion/android/settings/wear/views/SettingsWearTemplateTile.kt:69]: I'll make it blue to match the main settings page

File: common/src/main/java/io/homeassistant/companion/android/util/IntervalToString.kt [hunk 7]
```diff
@@ -0,0 +1,10 @@
+package io.homeassistant.companion.android.util
+
+fun IntervalToString(interval: Int): String {
+    return when {
+        interval == 0 -> "Manual"
+        interval >= 60 * 60 -> "${interval / 60 / 60} hours"
+        interval >= 60 -> "${interval / 60} minutes"
+        else -> "$interval seconds"
```
Comment (leroyboerefijn) [inline common/src/main/java/io/homeassistant/companion/android/util/IntervalToString.kt:8]: Need to use string resources here

**Review by JBassett — APPROVED** [review]

**Pull request merged**